js/views/stock.js

```diff
@@ -2,14 +2,17 @@
 function StockView() {
   $('#update-symbol').on('click', this.lookupSymbol.bind(this));
   $('#update-shares').on('click', this.updateShares.bind(this));
-}
+};
 
 StockView.prototype = {
   lookupSymbol: function() {
     var symbol = $("#symbol-input").val();
     this.model = new Stock(symbol);
-    this.render();
-    $('#cha-ching')[0].play();
+    this.model.refresh().then(function(){
+      this.render();
+      $('#cha-ching')[0].play();
+    }.bind(this));
+    console.log(symbol);
```

Comment:
this is a little finicky, but it goes back to the naming thing, ultimately i feel like were getting a stock here, looking up a symbol is just lines 9 and 10, but ultimately were getting the stock. This is also potentially a refactoring opportunity, where lines 9 and 10 could be in another function call.. but probably not, since you only use it once. The code is fine though, and honestly probably shouldn't be refactored. :+1: